app/src/androidTest/java/org/simple/clinic/overdue/AppointmentRepositoryAndroidTest.kt

```diff
@@ -118,64 +113,89 @@ class AppointmentRepositoryAndroidTest {
 
   @Test
   fun when_creating_new_appointment_then_the_appointment_should_be_saved() {
-    val patientId = UUID.randomUUID()
+    // given
     val appointmentDate = LocalDate.now(clock)
-    appointmentRepository.schedule(patientId, appointmentDate, Manual, testData.qaFacility()).blockingGet()
-
-    val savedAppointment = appointmentRepository.recordsWithSyncStatus(SyncStatus.PENDING).blockingGet().first()
-    savedAppointment.apply {
-      assertThat(this.patientUuid).isEqualTo(patientId)
-      assertThat(this.scheduledDate).isEqualTo(appointmentDate)
-      assertThat(this.status).isEqualTo(Scheduled)
-      assertThat(this.cancelReason).isEqualTo(null)
-      assertThat(this.syncStatus).isEqualTo(SyncStatus.PENDING)
+
+    //when
+    appointmentRepository.schedule(
+        patientUuid = patientUuid,
+        appointmentUuid = appointmentUuid,
+        appointmentDate = appointmentDate,
+        appointmentType = Manual,
+        currentFacility = facility
+    ).blockingGet()
+
+    // then
+    val savedAppointment = getAppointmentByUuid(appointmentUuid)
+    with(savedAppointment) {
+      assertThat(patientUuid).isEqualTo(this@AppointmentRepositoryAndroidTest.patientUuid)
+      assertThat(scheduledDate).isEqualTo(appointmentDate)
+      assertThat(status).isEqualTo(Scheduled)
+      assertThat(remindOn).isNull()
+      assertThat(cancelReason).isNull()
+      assertThat(agreedToVisit).isNull()
+      assertThat(syncStatus).isEqualTo(PENDING)
     }
   }
 
   @Test
-  fun when_creating_new_appointment_then_all_old_appointments_for_that_patient_should_be_canceled() {
-    val patientId = UUID.randomUUID()
-
-    val date1 = LocalDate.now(clock)
-    val timeOfSchedule = Instant.now(clock)
-    val facility = testData.qaFacility()
+  fun when_creating_new_appointment_then_all_old_appointments_for_that_patient_should_be_marked_as_visited() {
+    // given
+    val firstAppointmentUuid = UUID.fromString("0bc9cdb3-bfe9-41e9-88b9-2a072c748c47")
+    val scheduledDateOfFirstAppointment = LocalDate.parse("2018-01-01")
+    val firstAppointmentScheduledAtTimestamp = Instant.now(clock)
+    appointmentRepository.schedule(
+        patientUuid = patientUuid,
+        appointmentUuid = firstAppointmentUuid,
+        appointmentDate = scheduledDateOfFirstAppointment,
+        appointmentType = Manual,
+        currentFacility = facility
+    ).blockingGet()
+    markAppointmentSyncStatusAsDone(firstAppointmentUuid)
 
-    appointmentRepository.schedule(patientId, date1, Manual, facility).blockingGet()
+    testClock.advanceBy(Duration.ofHours(24))
```

Comment:
We also need to verify that the `createdAt` and `updatedAt` timestamps are modified correctly when executing the functions. I'll update the tests to make this more clear.

Reply:
Done.

Reply:
Yes, but we were verifying against a fixed value and not the `clock`.
Using `Instant.now(clock)` fixes this now.

Reply:
I don't understand? We need to verify when marking an older appointment as visited that the `createdAt` does not change, but the `updatedAt` does change. Since these instants are fetched from the `UtcClock` injected into the repository, we will have to advance the clock before running the method to be tested.

Reply:
The changes that you have made now, they are fine :)
I will discuss this in person later.